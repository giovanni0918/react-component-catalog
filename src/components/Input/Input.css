.Input {
    box-sizing: border-box;
    box-shadow: 0 2px 4px rgba(0, 0, 0, 0.3);
    font-family: Arial, Helvetica, sans-serif;
    font-size: 1rem;
    font-weight: 500;
    margin: 0;
    padding: 0.25rem;
    border: 1px solid lightskyblue;
    border-radius: 2px;
}

.Input~label {
    text-transform: capitalize;
}