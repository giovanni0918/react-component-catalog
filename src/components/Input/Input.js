// @flow
import React from 'react';
import styles from './Input.css';

type Props = {
  className: string,
  disabled: ?boolean,
  id: string,
  includeLabel: boolean,
  max: ?number,
  min: ?number,
  name: string,
  onBlur: ?Function,
  onChange: Function,
  onFocus: ?Function,
  required: ?boolean,
  step: ?number,
  type: string
};

const Input = ({
  className, disabled, id, includeLabel, max, min, name,
  onBlur, onChange, onFocus, required, step, type
}: Props) => (

    includeLabel ? (
      <div>
        <label htmlFor={id}>{`${name}: `}</label>
        <input
          className={className}
          disabled={disabled}
          id={id}
          max={max}
          min={min}
          name={name}
          onBlur={onBlur}
          onChange={onChange}
          onFocus={onFocus}
          placeholder={name}
          required={required}
          step={step}
          type={type}
        />
      </div>
    ) : (
        <input
          className={className}
          disabled={disabled}
          id={id}
          max={max}
          min={min}
          name={name}
          onBlur={onBlur}
          onChange={onChange}
          onFocus={onFocus}
          placeholder={name}
          required={required}
          step={step}
          type={type}
        />
      )
  );
export default Input;