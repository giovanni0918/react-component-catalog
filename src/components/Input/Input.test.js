import React from 'react';
import Input from './Input';
import renderer from 'react-test-renderer';

const SampleInput = (
  <Input
    className='Input'
    type='email'
    id='email'
    name='email'
    onChange={(event) => console.log('value', event.target.value)}
  />
);

test('It should always render an input', () => {
  const component = renderer.create(SampleInput);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});