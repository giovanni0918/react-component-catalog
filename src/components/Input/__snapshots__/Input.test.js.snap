// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It should always render an input 1`] = `
<input
  className="Input"
  id="email"
  name="email"
  onChange={[Function]}
  placeholder="email"
  type="email"
/>
`;
